src/WebJobs.Script.WebHost/Middleware/HostAvailabilityCheckMiddleware.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Script.WebHost.Features;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Middleware
+{
+    public partial class HostAvailabilityCheckMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly ILogger<HostAvailabilityCheckMiddleware> _logger;
+
+        public HostAvailabilityCheckMiddleware(RequestDelegate next, ILoggerFactory loggerFactory)
+        {
+            _next = next;
+            _logger = loggerFactory.CreateLogger<HostAvailabilityCheckMiddleware>();
+        }
+
+        public async Task Invoke(HttpContext httpContext, WebScriptHostManager manager)
+        {
+            using (Logger.VerifyingHostAvailabilityScope(_logger, httpContext.TraceIdentifier))
+            {
+                // in standby mode, we don't want to wait for host start
```

Comment:
Why? In standby mode we'll be serving an http warmup request which is a regular request to the placeholder site, right?

Reply:
This was just to stay consistent with the logic we had in V1. I agree that it's odd to special case this here, so would be glad to remove this.

Reply:
In v1 the warmup request ultimately gets routed through the FunctionsController which has an imperative call to DelayUntilHostReady, right? It has to, because the request is going to be routed to an http trigger function, so the host has to be up.

Reply:
I'll remove this check here.